source/features/enable-file-links-in-compare-view.tsx

```diff
@@ -0,0 +1,82 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import GitHubURL from '../github-helpers/github-url';
+
+function handleIsPRMenuOpening(event: delegate.Event): void {
+	const dropdown = event.delegateTarget.nextElementSibling!;
+	event.delegateTarget.classList.add('rgh-actionable-link'); // Mark this as processed
+
+	// Only enabled on Open/Draft PRs. Editing files doesn't make sense after a PR is closed/merged.
+	if (!select.exists('.gh-header-meta [title$="Open"], .gh-header-meta [title$="Draft"]')) {
+		return;
+	}
+
+	// If you're viewing changes from partial commits, ensure you're on the latest one.
+	const isPartialCommits = select.exists('.js-commits-filtered');
+	if (isPartialCommits && !select.exists('[aria-label="You are viewing the latest commit"]')) {
+		return;
+	}
+
+	// This solution accounts for:
+	// - Branches with slashes in it
+	// - PRs opened from the default branch
+	const headBranchUrl = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname;
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const filepath = dropdown.closest<HTMLDivElement>('[data-path]')!.dataset.path;
+	viewFile.pathname = headBranchUrl + '/' + String(filepath);
+}
+
+function handleIsCompareMenuOpening(event: delegate.Event): void {
+	const dropdown = event.delegateTarget.nextElementSibling!;
+	event.delegateTarget.classList.add('rgh-actionable-link'); // Mark this as processed
+
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const url = new GitHubURL(viewFile.href);
+	url.assign({
+		branch: location.pathname.replace(/.+:|.+\.{3}/, '')
+	});
+	viewFile.href = String(url);
+
+	// Dont replace the edit and delete buttons if its not possible
+	if (!select.exists([
+		'[name="collab_privs"]', // Excludes repo's that you dont have rights to edit and Permalink's (the whole create pull request will not exist on a Permalink)
+		'.js-issue-sidebar-form' // Dont exclude same repo compare pages you have rights to by detecting if the review's/labels exists
+	])) {
+		return;
+	}
```

Comment:
Doesn't this apply to View links as well?
The feature "fixes" the View link by making Edit/Delete possible on the View page.
If Edit/Delete isn't possible here, it won't be possible on the next page either.

I think this condition should be moved to `features.add`’s `exclude`. I think this also applies to the 2 conditions in `handleIsPRMenuOpening`

Also please improve your comments, there are a lot of typos (It's `repos` or `repositories`, not `repo's`). The lack of quotation marks, capitalization and consistency also make them hard to read.